src/coreComponents/mpiCommunications/NeighborCommunicator.hpp

```diff
@@ -39,16 +35,14 @@ inline int CommTag( int const GEOSX_UNUSED_PARAM( senderRank ),
 }
 
 class MeshLevel;
-class ObjectManagerBase;
 class NeighborCommunicator
 {
 public:
 
-  NeighborCommunicator();
+  explicit NeighborCommunicator( int rank );
 
-  NeighborCommunicator( int rank ):
-    NeighborCommunicator()
-  { setNeighborRank( rank ); }
+  NeighborCommunicator():
```

Comment:
Does it even need to be default construct-able? I would say it probably shouldn't be.

Reply:
Actually yes it does since we have a `std::vector< NeighborCommunicator >` which is managed through `Group` and `Wrappers`. This induces some default constructor constraint for some STL stuff (`resize` if I can read correctly).

But this is indeed utility constraint and not business and we were close to be able to remove it. 🤷